sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -160,4 +166,12 @@ class BooleanSimplificationSuite extends PlanTest with PredicateHelper {
       testRelation.where('a > 2 || ('b > 3 && 'b < 5)))
     comparePlans(actual, expected)
   }
+
+  test("Complementation Laws") {
+    checkCondition('a && !'a, LocalRelation(testRelation.output, Seq.empty))
```

Comment:
ah i see. Then we need to create a new test relation with data, and test if the result is a empty relation here